pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -561,3 +586,67 @@ func runCDCMixedVersions(ctx context.Context, t test.Test, c cluster.Cluster) {
 	mvt.AfterUpgradeFinalized("wait and validate", tester.waitAndValidate)
 	mvt.Run()
 }
+
+// runCDCMixedVersionCheckpointing tests that the new span-level checkpoint
+// code added in 25.2 works correctly in mixed-version states.
+//
+// Writing the checkpoint is explicitly forced with cluster settings and
+// restoring the checkpoint implicitly happens following each of the rolling
+// restarts during the mixed-version test run.
+func runCDCMixedVersionCheckpointing(ctx context.Context, t test.Test, c cluster.Cluster) {
+	tester := newCDCMixedVersionTester(ctx, c)
+
+	mvt := mixedversion.NewTest(
+		ctx, t, t.L(), c, tester.crdbNodes,
+		// We're only concerned with mixed-version compatibility starting at
+		// versions that can upgrade to 25.2 (only 24.3 and 25.1), since that's
+		// the first version with the new span-level checkpoint format.
+		mixedversion.MinimumBootstrapVersion("v24.3.0"),
```

Comment:
nit: can you use `mixedversion.MinimumSupportedVersion` instead? The extra upgrades shouldn't take that long, and if they do you can use `mixedversion.MaxNumPlanSteps` to limit run time (which is better since it won't rot when we bump the current version). 

`mixedversion.MinimumBootstrapVersion` is more intended for the rare case you need the cluster to start+init on a certain version. And as you already ran into, has a few bugs 😃